File: td.py
import pandas as pd 
import seaborn as srn 
import statistics as sts

dataset = pd.read_csv("tempo.csv", sep=";")
#explora o df categorico
agrup = dataset.groupby(['Vento']).size()

#mosta quais colunas tem valores nulos
print(dataset.isnull().sum())

#UMIDADE----------
#preenche os campos nulos com a media e mediana
medianav = sts.median(dataset['Umidade'])
# Preenche os valores nulos da coluna 'Umidade' com a mediana
dataset['Umidade'].fillna(medianav, inplace=True)
#preenche os valores maiores que 200 da coluna de umidade com a mediana
dataset.loc[dataset['Umidade'] > 100, 'Umidade'] = medianav

#TEMPERATURA---------
#mediana de temperatura
medianat = sts.median(dataset['Temperatura'])
#preenche o valor maior que 100 com a mediana
dataset.loc[dataset['Temperatura'] > 100, 'Temperatura'] = medianat

#VENTO----------
# Preenche os valores nulos da coluna 'Vento' com a moda
dataset['Vento'].fillna("FALSO", inplace=True)

#APARENCIA---------
#atribuindo  a moda de aparencia em case fora de categoria
agrupo = dataset.groupby(['Aparencia']).size()
print(agrupo)
#categorico, escolho os que ta fora de ordem e atribuo a moda da coluna aparencia
dataset.loc[dataset['Aparencia'].isin(['menos']), 'Aparencia'] = "chuva"

#JOGAR--------
#Preenche os valores de jogar de acordo com a aparencia
dataset.loc[dataset['Aparencia'] == "chuva", 'jogar'] = "nao"
dataset.loc[dataset['Aparencia'] == "sol", 'jogar'] = "sim"
dataset.loc[dataset['Aparencia'] == "nublado", 'jogar'] = "sim"

#DATASET------- (FINALMENTE)
print(dataset)
# Salva o DataFrame limpo em um novo arquivo CSV
dataset.to_csv("tempo_limpo2.csv", index=False)

File: td.r
```{r}
# Limpeza de dados do tempo
#CARREGA OS DADOS
    dados = read.csv("tempo.csv", sep = ";", na.string = c(" ","", NA), stringsAsFactors=FALSE)
    
    # PROCURA OS DADOS NA E DADOS MAIORES QUE 90, ADICONA A MEDIANA NESSES DADOS(COLUNA UMIDADE)
    dados$Umidade[is.na(dados$Umidade)|dados$Umidade >90] <- median(dados$Umidade, na.rm=TRUE)
    
    #ADICONA A MODA NO NAS DA COLUNA VENTO
    dados$Vento[is.na(dados$Vento)] <- FALSE
    
    #ADICIONA A MEDIANA NO NAS DA COLUNA TEMPERATURA, CONVERTE PARA CELSIUS E SUBSTITUI OS VALORES MAIORES QUE 100 POR MEDIANA
    dados$Temperatura[dados$Temperatura > 100] <- median(dados$Temperatura[dados$Temperatura <= 100])
    dados$Temperatura <- (dados$Temperatura - 32) * 5/9
    
    #ADICIONA A MODA NO NAS DA COLUNA APARENCIA
    dados$Aparencia[!dados$Aparencia %in% c("chuva","nublado","sol")]<- "chuva"
    dados
    write.csv(dados, "tempo_limpo.csv", row.names = FALSE)

```